tests/unit/pyrflx_test.py

```diff
@@ -1203,3 +1204,93 @@ def test_always_valid_aspect(  # pylint: disable=invalid-name
     assert message_always_valid_aspect_value.get("F1") == "RFLX_UNKNOWN_ENUM_1"
     assert message_always_valid_aspect_value.get("F2") == "RFLX_UNKNOWN_ENUM_2"
     assert message_always_valid_aspect_value.get("F3") == 171
+
+
+def test_get_inner_messages(
+    array_message_package: Package, message_array_refinement_value: MessageValue
+) -> None:
+    array_message_one = array_message_package["Array_Member"]
+    array_message_one.set("Byte", 5)
+    array_message_two = array_message_package["Array_Member"]
+    array_message_two.set("Byte", 6)
+    sdu_message = array_message_package["Array_Member"]
+    sdu_message.set("Byte", 7)
+
+    message_foo_value = message_array_refinement_value
+    array_contents: Sequence[TypeValue] = [array_message_one, array_message_two]
+    message_foo_value.set("Length", 2)
+    message_foo_value.set("Array_Field", array_contents)
+    message_foo_value.set("Payload", sdu_message.bytestring)
```

Comment:
Shouldn't "Payload" be settable with just `sdu_message` when a refinement for "Payload" is defined?

Reply:
Not currently. The value must be `value: Union[bytes, int, str, Sequence[TypeValue], Bitstring]`. The assigned inner message is parsed from bytes in `parse()` of `OpaqueValue`. If that should be changed, I would do so in a new issue.

Reply:
I see, that means refinements are only used for parsing, but not for serialization in PyRFLX. I think there is no need to change that now.